### PR TITLE
Docusaurus v1 site

### PR DESCRIPTION
**Pull request by Burnett2k**
### Description

New [Vercel Site](https://react-swipeable-git-docusaurus-v1-formidable-labs.vercel.app/)

Existing [github pages site](https://formidablelabs.github.io/react-swipeable/)

[Search plugin added](https://github.com/easyops-cn/docusaurus-search-local?tab=readme-ov-file)

Adding a new Docusaurus Documentation site. The content of the new site has been extracted from the already excellently written `README.md` file and put into the /docs portion of the site.

I also got the examples from the example app working within the documentation. However, I left the old examples repo in place in case users want to run locally instead.

Things left to be done (open to suggestions)

## Landing Page

- [x] use different images / icons for features on landing page
- [x] use new color palette
- [x] update to use banner and look more like nuka-carousel
- [x] add "more from Nearform Commerce" section
- [x] clean up verbiage / terminology on newly created pages
- [x] get feedback from design / others (in progress)

## Documentation site
- [x] clean up verbiage / terminology on newly created pages
- [x] convert to using static images in carousels for faster loading

## Other
- [ ] Note: `react-swipeable` doesn't currently work with node 18, so you need to be on 16 when running the demo app locally. However, the docs site (using Docusaurus framework) requires at least node 18, so you have to currently switch between node versions. I can work on fixing this. 🚧 Will do this in a separate pr 🚧 
- [x] It appears there's some CICD checks using node 12 which might fail

#### Type of Change

Documentation 

### How Has This Been Tested?

* Running locally
* Tested on the Vercel preview site

mobile:
image

api docs:
image

working demo:
image

### Checklist: (Feel free to delete this section upon completion)

- [x] I have included a changeset if this change will require a version change to one of the packages.
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have run all builds, tests, and linting and all checks pass
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] My changes generate no new warnings
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by hartzis** [comment]
this is awesomeness! always wanted this package to have a proper doc site 😸🙌❤️

**Comment by Burnett2k** [comment]
> this is awesomeness! always wanted this package to have a proper doc site 😸🙌❤️

Thanks @hartzis . I'm going to keep polishing it up a bit. Curious if you have any thoughts for improvement? I was considering updating the examples to use a different carousel potentially that we can just pull straight from npm, or even `nuka-carousel` for some cross-promotion.

File: docs/src/components/landing/hero.tsx [hunk 1]
```diff
@@ -0,0 +1,14 @@
+import React from "react";
+import { ProjectBadge } from "formidable-oss-badges";
+
+export default function Hero(): JSX.Element {
+  return (
+    <div>
+      <ProjectBadge
+        color="#190E"
```
Comment (BrittanyIRL) [inline docs/src/components/landing/hero.tsx:8]: for the sake of the draft - thoughts on running with #37B756 as the green here? it ties to the feature assets that are getting used below and is at least a jumping-off point for continuity.

Comment (Burnett2k) [reply docs/src/components/landing/hero.tsx:8]: Yea, I'm open to changing the color for sure. Actually, @AlexSwanson pointed out there's a blue color in the Figma designs specifically for react swipeable, so I'm going to build a palette off of that.

image

Comment (Burnett2k) [reply docs/src/components/landing/hero.tsx]: @BrittanyIRL I've updated to use new colors based off of this image. When you get a chance, take a look and let me know what you think. I'm ok with updating or having design provide better assets. I did what I could with Figma, but admittedly it's not my area of expertise.

File: docs/docusaurus.config.ts [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+          className: "header-github-link",
+          position: "right",
+        },
+      ],
+    },
+    footer: {
+      logo: {
+        alt: "Nearform logo",
+        src: "img/nearform-logo-white.svg",
+        href: "https://nearform.com",
```
Comment (BrittanyIRL) [inline docs/docusaurus.config.ts:82]:
```suggestion
        href: "https://commerce.nearform.com",
```

File: docs/src/pages/index.tsx [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+      </div>
+      <LandingFeatures
+        heading="Features"
+        list={[
+          {
+            imgSrc: feature1,
+            alt: "robust",
+            title: "Robust",
+            html: {
+              __html:
```
Comment (BrittanyIRL) [inline docs/src/pages/index.tsx:43]: 🤔 what can go here aside from the double mention of `useSwipeable` as 2 of 3 features? In your update did anything else come to mind that could be used here?

**Comment by BrittanyIRL** [comment]
Lookin' good! Found a couple things, flagged 'em in vercel comments mostly. one of them might also be on Nuka and I just missed it. 

Good catch with the dueling versions for usage. I think that would be a great thing to work on if you find yourself blocked by needing review here tomorrow. Holler when you're ready for a second pass. Looks so great!

**Comment by Burnett2k** [comment]
> Lookin' good! Found a couple things, flagged 'em in vercel comments mostly. one of them might also be on Nuka and I just missed it.
> 
> Good catch with the dueling versions for usage. I think that would be a great thing to work on if you find yourself blocked by needing review here tomorrow. Holler when you're ready for a second pass. Looks so great!

Thanks for checking. The only remaining issue right now is with the Copy button looking weird on Safari. I'll keep working on that tomorrow. I left the other vercel comment open for you to review and close if you think it's fixed on your iPhone 13 mini. From what I can tell, it seems fixed.

The node version issue shouldn't be a blocker for releasing this. It's just kind of confusing, but shouldn't prevent the PR from being approved and merged as I was able to get CI working by putting in a couple of other fixes.

However, I do think we should do a follow-up PR to get node18 working which will probably require a major or minor release depending on what we have to do.

**Comment by BrittanyIRL** [comment]
Here's a fix for the buttons: https://github.com/FormidableLabs/nuka-carousel/pull/1046/commits/227510944bbdb94ca0c273bbec9e6c073ca095a7 

Thanks for clarifying, I don't mean that the node update needs to be part of this update I just mean that it's a valuable update that isn't blocked by review of this branch since it's its own thing :)

**Review by hartzis — APPROVED** [review]
🙌 woo woo!

📓 on the node upgrade: we should really try to avoid a major bump for a dev dep unless there is something fundamentally changing in the output which i don't think should be the case.

**Review by BrittanyIRL — APPROVED** [review]
Thanks for the changes!

**Pull request merged**